File: 2 - Poseidon's Path Score/solution.js
function calcPathScore(s) {
    let ans = 0;
    for (let i = 0; i < s.length - 1; i++) {
        ans += Math.abs(s.charCodeAt(i) - s.charCodeAt(i + 1));
    }
    return ans;
}

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question('', t => {
    let count = parseInt(t);
    let inputs = [];
    readline.on('line', line => {
        inputs.push(line);
        if (inputs.length === count) {
            inputs.forEach(s => console.log(calcPathScore(s)));
            readline.close();
        }
    });
});

File: 2 - Poseidon's Path Score/solution.py
def calc_path_score(s):
    ans = 0
    for i in range(len(s) - 1):
        ans += abs(ord(s[i]) - ord(s[i + 1]))
    return ans

t = int(input())

for _ in range(t):
    s = input().strip()
    print(calc_path_score(s))
